types/google-cloud__storage/index.d.ts

```diff
@@ -5,7 +5,340 @@
+        createChannel(id: string, config: ChannelConfig): Promise<[Channel, Storage.ApiResponse]>;
+        delete(): Promise<[Storage.ApiResponse]>;
+        deleteFiles(query?: BucketQuery): Promise<void>;
+        exists(): Promise<[boolean]>;
+        file(name: string, options?: BucketFileOptions): File;
+        get(options?: BucketGetOptions): Promise<[Bucket, Storage.ApiResponse]>;
+        getFiles(query?: BucketQuery): Promise<[File[]]>;
+        getFilesStream(query?: BucketQuery): ReadStream;
+        getMetadata(): Promise<[BucketMetadata, Storage.ApiResponse]>;
+        id: string;
+        makePrivate(options?: BucketPrivacyOptions): Promise<[File[]]>;
+        makePublic(options?: BucketPrivacyOptions): Promise<[File[]]>;
+        metadata: BucketMetadata;
+        name: string;
+        setMetadata(metadata?: BucketMetadata): Promise<[Storage.ApiResponse]>;
+        upload(localPath: string, options?: Storage.UploadOptions): Promise<[File]>;
+    }
+
+    /**
+     * Bucket configuration.
+     */
+    interface BucketConfig {
+        coldline?: boolean;
+        dra?: boolean;
+        location?: string;
+        multiRegional?: boolean;
+        nearline?: boolean;
+        regional?: boolean;
+        versioning?: {
+            enabled?: boolean
+        };
+    }
+
+    /**
+     * Options for getting a file.
+     */
+    interface BucketFileOptions {
+        generation?: string | number;
+    }
+
+    /**
+     * Options for getting a bucket.
+     */
+    interface BucketGetOptions {
+        autoCreate?: boolean;
+    }
+
+    /**
+     * Bucket metadata.
+     */
+    interface BucketMetadata {
+        etag?: string;
+        id?: string;
+        kind?: string;
+        location?: string;
+        metageneration?: string;
+        name?: string;
+        projectNumber?: string;
+        selfLink?: string;
+        storageClass?: string;
+        timeCreated?: string;
+        updated?: string;
+    }
+
+    /**
+     * The options for making the bucket private.
+     */
+    interface BucketPrivacyOptions {
+        includeFiles?: boolean;
+        force?: boolean;
+    }
+
+    /**
+     * Query a bucket.
+     */
+    interface BucketQuery {
+        autoPaginate?: boolean;
+        delimiter?: string;
+        prefix?: string;
+        maxApiCalls?: number;
+        maxResults?: number;
+        pageToken?: string;
+        versions?: boolean;
+    }
+
+    /**
+     * Options for specifying the content length range.
+     */
+    interface ContentLengthRange {
+        max?: number;
+        min?: number;
+    }
+
+    /**
+     * A file in the cloud.
+     */
+    class File {
+        constructor(bucket: Bucket, name: string, options?: BucketFileOptions);
+        acl: Acl;
+        copy(destination: string | Bucket | File): Promise<[File, ApiResponse]>;
+        createReadStream(options?: ReadStreamOptions): ReadStream;
+        createWriteStream(options?: WriteStreamOptions): WriteStream;
```

Comment:
[`createResumableUpload`](https://googlecloudplatform.github.io/google-cloud-node/#/docs/storage/1.2.0/storage/file?method=createResumableUpload) seems to be missing.